cluster-agent/controllers/argoproj.io/namespace_reconciler_test.go

```diff
@@ -465,14 +467,285 @@ var _ = Describe("Namespace Reconciler Tests.", func() {
 			err := k8sClient.Create(ctx, &secret)
 			Expect(err).To(BeNil())
 
-			By("Call runSecretCleanup function.")
+			By("Call cleanOrphanedCRsfromCluster_Secret function.")
 
-			runSecretCleanup(ctx, dbq, k8sClient, log)
+			cleanOrphanedCRsfromCluster_Secret(ctx, dbq, k8sClient, log)
 
 			By("Verify cluster secret from cluster is deleted.")
 
 			err = k8sClient.Get(ctx, client.ObjectKeyFromObject(&secret), &secret)
 			Expect(err).To(BeNil())
 		})
 	})
+
+	Context("Testing for cleanOrphanedCRsfromCluster_Operation function.", func() {
+		var log logr.Logger
+		var ctx context.Context
+		var dbq db.AllDatabaseQueries
+		var k8sClient client.WithWatch
+		var k8sOperationFirst managedgitopsv1alpha1.Operation
+		var gitopsEngineInstance *db.GitopsEngineInstance
+
+		BeforeEach(func() {
+			err := db.SetupForTestingDBGinkgo()
+			Expect(err).To(BeNil())
+
+			ctx = context.Background()
+			log = logger.FromContext(ctx)
+			dbq, err = db.NewUnsafePostgresDBQueries(true, true)
+			Expect(err).To(BeNil())
+
+			scheme, _, _, _, err := tests.GenericTestSetup()
+			Expect(err).To(BeNil())
+
+			err = appv1.AddToScheme(scheme)
+			Expect(err).To(BeNil())
+
+			// Fake kube client.
+			k8sClient = fake.NewClientBuilder().WithScheme(scheme).Build()
+
+			_, _, _, gitopsEngineInstance, _, err = db.CreateSampleData(dbq)
+			Expect(err).To(BeNil())
+
+			By("Create Secret CR.")
+
+			// Create K8s operation CR
+			k8sOperationFirst = managedgitopsv1alpha1.Operation{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-op-" + string(uuid.NewUUID()),
+					Namespace: "test-ns-1",
+				},
+				Spec: managedgitopsv1alpha1.OperationSpec{
+					OperationID: "test-id-" + string(uuid.NewUUID()),
+				},
+			}
+		})
+
+		It("Should delete Operation if it doesn't point to a DB entry.", func() {
+			k8sOperationSecond := k8sOperationFirst
+
+			// Create 1st CR in cluster
+			err := k8sClient.Create(ctx, &k8sOperationFirst)
+			Expect(err).To(BeNil())
+
+			// Create 2nd CR in cluster
+			k8sOperationSecond.Name = "test-op-" + string(uuid.NewUUID())
+			k8sOperationSecond.Spec.OperationID = "test-id-" + string(uuid.NewUUID())
+			err = k8sClient.Create(ctx, &k8sOperationSecond)
+			Expect(err).To(BeNil())
+
+			By("Calling cleanOrphanedCRsfromCluster_Operation function to delete orphaned Operation CR, if corresponding DB entry is not present.")
+
+			cleanOrphanedCRsfromCluster_Operation(ctx, dbq, k8sClient, log)
+
+			By("Verify that orphaned Operation CRs without a DB entry are deleted.")
+
+			err = k8sClient.Get(ctx, types.NamespacedName{Name: k8sOperationFirst.Name, Namespace: k8sOperationFirst.Namespace}, &k8sOperationFirst)
+			Expect(apierr.IsNotFound(err)).To(BeTrue())
+
+			err = k8sClient.Get(ctx, types.NamespacedName{Name: k8sOperationSecond.Name, Namespace: k8sOperationSecond.Namespace}, &k8sOperationSecond)
+			Expect(apierr.IsNotFound(err)).To(BeTrue())
+		})
+
+		It("Should not delete an Operation if it points to a valid DB entry.", func() {
```

Comment:
How does this sound: `Should not delete an Operation if it points to a valid DB entry` => `Should not delete an Operation if it points to a valid DB entry, as long as the operation is waiting`